common/common.go

```diff
@@ -38,11 +38,14 @@ func InitHoneycombFromEnvVars() error {
 	sampleRate = 1
 	if os.Getenv("SAMPLE_RATE") != "" {
 		i, err := strconv.Atoi(os.Getenv("SAMPLE_RATE"))
-		if err != nil {
+		// including a check on sample rate being configured as 0.
+		// assumption: intention of setting 0 is that no sampling is desired.
+		if err != nil || uint(i) == 0 {
```

Comment:
I think we should use <= here. Casting a negative number into a `uint` doesn't result in `0`. For example: [Go Playground](https://go.dev/play/p/7MXQ7W0YhNu)

```suggestion
		if err != nil || i <= 0 {
```